Sources/OpenAI/Public/Models/AudioSpeechResult.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  AudioSpeechResult.swift
+//
+//
+//  Created by Ihor Makhnyk on 13.11.2023.
+//
+
+import Foundation
+#if canImport(AVFoundation)
+import AVFoundation
+#endif
+
+public struct AudioSpeechResult {
+    
+    /// Audio data for one of the following formats :`mp3`, `opus`, `aac`, `flac`
+    public let audioData: Data?
+    
+    /// Saves the audio data to a file at a specified file path.
+    ///
+    /// - Parameters:
+    ///     - name: The name for the file.
+    ///     - format: The format of the audio data, as defined in **`AudioSpeechQuery.AudioSpeechResponseFormat`**.  For example: **`.mp3`**
+    ///     - path: The destination file path as an URL.
+    /// - Throws: Throws an NSError if there is an issue with writing the data to the specified file path.
+    public func saveAs(_ name: String, format: AudioSpeechQuery.AudioSpeechResponseFormat, to path: URL) throws {
```

Comment:
Why do we need to have `saveAs` method inside `AudioSpeechResult` struct? I need clarification on it.

Reply:
Well, to enable devs conveniently be able to save the data to file, as we only receive a raw audio data to file, so unless we use some kind of player that supports raw data, we have to save it as a file. It may be sometimes not obvious what will the request return, so mostly it is file or url to file, but this is the only case where we have raw data, that was my motive to have more flexibility built in.